### ISSUE TITLE
Request for guides to be deployed for mobility

### ISSUE THREAD

**Issue by evaristoc** [freeCodeCamp/guides#20]
@raiseadead @QuincyLarson @Bouncey 

A discussion in CT started about deployment for mobility of FCC resources. "Guides" was suggested as target.

This post opens a discussion about the desirability and feasibility of this request and if decided to start working on a road map.

**Comment by QuincyLarson** [comment]
@evaristoc yes, we plan to eventually get Index.db working, or to shell this in a mobile app so everything can be read offline. That is definitely a goal of ours!

**Comment by Bouncey** [comment]
Would the service workers do what we need on this?

We do not query a db to pull in the articles.

**Comment by QuincyLarson** [comment]
@Bouncey Good question. IndexDB would probably be a poor fit for this, then. Is there a library we could use to cache all the articles in the browser so it can be used offline?

**Comment by evaristoc** [comment]
https://redis.io/topics/persistence? They mention some pro's and con's in the reference.

**Comment by raisedadead** [comment]
@evaristoc I think we would want client side peristance. @QuincyLarson I think a PWA is the best candidate for this. 

If you are wondering, then if you browse to our forum (which is on discourse) from a phone, and then turn off the data. You can still have the content auto-magically in there including posts.

On similar lines you should check out:
https://wiki-offline.jakearchibald.com

That's an awesome example of what we need.

**Comment by QuincyLarson** [comment]
@raisedadead Awesome - do you know of a way we can get Gatsby to behave like a PWA? We are also planning to make the Guide compatible with Google AMP and Facebook Instant Articles.

**Comment by evaristoc** [comment]
@raisedadead Interesting! @QuincyLarson shouldn't we contact Gatsby's creator?

**Comment by Bouncey** [comment]
[gatsby-plugin-offline](https://www.gatsbyjs.org/packages/gatsby-plugin-offline/)

This plugin adds a service worker

**Comment by QuincyLarson** [comment]
@Bouncey Awesome! Do you know whether this will work when we deploy to Netlify?

**Comment by Bouncey** [comment]
There is no reason it shouldn't. We would need an offline search page though

**Comment by QuincyLarson** [comment]
@Bouncey Awesome! When you say "offline search page" you mean something that runs searches based off of articles that's cached in the browser? How much work do you think that would be? If it would be too much work, I propose we just disable search while offline for now so we can focus on shipping.

**Comment by Bouncey** [comment]
That is possible yes, but for now I meant just some UI explaining why the search would not be working.

**Comment by QuincyLarson** [comment]
@Bouncey OK - that sounds good. I think disabling the search box and filling it in with "You're currently offline" would work fine.

**Comment by thecodingaviator** [comment]
Guys, @Bouncey , @QuincyLarson , any update on this?
